src/v/cluster/metadata_dissemination_service.cc

```diff
@@ -145,18 +146,24 @@ ss::future<> metadata_dissemination_service::start() {
 
 void metadata_dissemination_service::handle_leadership_notification(
   model::ntp ntp, model::term_id term, std::optional<model::node_id> lid) {
-    (void)ss::with_gate(_bg, [this, ntp = std::move(ntp), lid, term]() mutable {
-        // the lock sequences the updates from raft
-        return _lock.with([this, ntp = std::move(ntp), lid, term]() mutable {
-            return container().invoke_on(
-              0,
-              [ntp = std::move(ntp), lid, term](
-                metadata_dissemination_service& s) mutable {
-                  return s.apply_leadership_notification(
-                    std::move(ntp), term, lid);
-              });
-        });
-    });
+    try {
+        (void)ss::with_gate(
+          _bg, [this, ntp = std::move(ntp), lid, term]() mutable {
+              // the lock sequences the updates from raft
+              return _lock.with(
+                [this, ntp = std::move(ntp), lid, term]() mutable {
+                    return container().invoke_on(
+                      0,
+                      [ntp = std::move(ntp), lid, term](
+                        metadata_dissemination_service& s) mutable {
+                          return s.apply_leadership_notification(
+                            std::move(ntp), term, lid);
+                      });
+                });
+          });
+    } catch (const ss::gate_closed_exception&) {
+        // Should ignore this exception, because it signal about shutdown
+    }
```

Comment:
@VadimPlh @jcsp 

It isn't clear from the commit if this is attempting to handle _entering_ the gate when its closed, or ignoring the gate closed exception that occurs within the background fiber.

The code here handles the former case (which could also be checked inline with `gate::is_closed`), but the later case is not handled and will result in an unhandled exceptional future error.

You can use `ss::try_with_gate` instead which will not throw immediately, but instead return an exceptional future if the gate is already closed. This allows the handling of both cases to be unified around `future::handle_exception_type`.